src/jep/InvocationHandler.java

```diff
@@ -40,6 +40,7 @@
     private long target;
 
     private Jep jep;
+    private final boolean isDirect;
```

Comment:
Thank you for the explanation.  I'd like to not have "is" in the variable name, Java standard is to have the getter for booleans be isVariableName().  Since direct wasn't immediately clear to me, a better name could be functionalInterface (and getter isFunctionalInterface()).  Since that's essentially what is happening here even if we're not supporting Java 8 language features yet.